File: src/main/kotlin/model/JobApplication.kt
package com.example.model

import kotlinx.serialization.Serializable

@Serializable
enum class ApplicationStatus{
    APPLIED,
    INTERVIEW,
    REJECTED,
    WITHDRAWN,
    OFFER
}



@Serializable
data class JobApplication(
    val ownerId: String,
    val id : String,
    val company: String,
    val status: ApplicationStatus,
    val title: String,
    val appliedAt: String,
    val description: String? = null,
    val link: String? = null,
    val city: String? = null,
)

File: frontend/src/features/applications/model/jobApplication.ts
export type ApplicationStatus =
  | "APPLIED"
  | "INTERVIEW"
  | "REJECTED"
  | "WITHDRAWN"
  | "OFFER";

export interface StatusTransition {
  status: ApplicationStatus;
  changedAt: string;
}

export interface JobApplication {
  id: string;
  company: string;
  status: ApplicationStatus;
  title: string;
  appliedAt: string;
  description?: string | null;
  link?: string | null;
  city?: string | null;
  statusHistory: StatusTransition[];
}
